src/libraries/System.Private.Xml/src/System/Xml/XmlUrlResolver.cs

```diff
@@ -42,14 +38,25 @@ public RequestCachePolicy CachePolicy
         // Maps a URI to an Object containing the actual resource.
         public override object? GetEntity(Uri absoluteUri, string? role, Type? ofObjectToReturn)
         {
-            if (ofObjectToReturn is null || ofObjectToReturn == typeof(System.IO.Stream) || ofObjectToReturn == typeof(object))
+            if (ofObjectToReturn is null || ofObjectToReturn == typeof(Stream) || ofObjectToReturn == typeof(object))
             {
                 return XmlDownloadManager.GetStream(absoluteUri, _credentials, _proxy);
             }
 
             throw new XmlException(SR.Xml_UnsupportedClass, string.Empty);
         }
 
+        // Maps a URI to an Object containing the actual resource.
+        public override async Task<object> GetEntityAsync(Uri absoluteUri, string? role, Type? ofObjectToReturn)
```

Comment:
I moved this without changes, for some reason we had separate file for async version (same for XmlDownloadManager)